packages/contracts-bedrock/test/scripts/L2Genesis.t.sol

```diff
@@ -0,0 +1,242 @@
+        super.setUp();
+        Artifacts.setUp();
+
+        string memory deployConfigPath = string.concat(vm.projectRoot(), "/deploy-config/", deploymentContext, ".json");
+        if (!vm.exists(deployConfigPath)) {
+            revert(
+                string.concat(
+                    "Did not find deploy config at: ",
+                    deployConfigPath
+                )
+            );
+        }
+
+        vm.etch(address(cfg), vm.getDeployedCode("DeployConfig.s.sol:DeployConfig"));
+        vm.label(address(cfg), "DeployConfig");
+        vm.allowCheatcodes(address(cfg));
+        cfg.read(deployConfigPath);
+
+        genesisPath = string.concat(vm.projectRoot(), "/deployments/", deploymentContext, "/genesis-l2.json");
+        if (!vm.exists(genesisPath)) {
+            string[] memory commands = new string[](3);
+            commands[0] = "bash";
+            commands[1] = "-c";
+            commands[2] = string.concat(
+                "DEPLOYMENT_CONTEXT=",
+                deploymentContext,
+                " forge script --chain-id ",
+                vm.toString(cfg.l2ChainID()),
+                " ./scripts/L2Genesis.s.sol:L2Genesis"
+            );
+            vm.ffi(commands);
+        }
+    }
+
+    /// @notice Iterates over every alloc parsed from `genesisPath`, and depending on
+    ///         the alloc address, runs specific checks based on whether the alloc is
+    ///         a precompile, predeploy proxy, or predeploy implementation. If the
+    ///         alloc address doesn't pass any of the checks to determine what it is,
+    ///         the function reverts with.
+    function test_allocs() external {
+        Alloc[] memory allocs = _parseAllocs(genesisPath);
+
+        for (uint256 i; i < allocs.length; i++) {
+            uint160 numericAddress = uint160(allocs[i].addr);
+            if (numericAddress < L2GenesisHelpers.PRECOMPILE_COUNT) {
+                _checkPrecompile(allocs[i]);
+            } else if (_isProxyAddress(allocs[i].addr)) {
+                _checkProxy(allocs[i]);
+            } else if (_isImplementationAddress(allocs[i].addr)) {
+                _checkImplementation(allocs[i]);
+            } else if (_isDevAccount(allocs[i].addr)) {
+                _checkDevAccount(allocs[i]);
+            } else {
+                revert(string.concat("Unknown alloc: ", vm.toString(allocs[i].addr)));
+            }
+        }
+    }
+
+    function test_setPredeployProxies() external {
+        // vm.record();
+        // l2Genesis._setPredeployProxies();
+        // bytes32[] memory reads;
+        // bytes32[] memory writes;
+        // (reads, writes) = vm.accesses(0x4200000000000000000000000000000000000000);
+
+        // console.log(reads.length);
+        // console.log(writes.length);
+        // for (uint i; i < writes.length; i++) {
+        //     console.logBytes32(writes[i]);
+        // }
+
+        vm.startStateDiffRecording();
+        l2Genesis._setPredeployProxies();
```

Comment:
I don't think this is working bc you aren't calling `setup()` on the `L2Genesis` contract

Reply:
Also you should remove the underscore prefixes if you make the functions public

Reply:
> I don't think this is working bc you aren't calling setup() on the L2Genesis contract

As discussed in Slack thread, we think this is a limitation of the cheatcode not including `vm.etch`es and `vm.store`s

Reply:
This was solved right?